chain/network/src/peer/peer_actor.rs

```diff
@@ -545,6 +546,82 @@ impl PeerActor {
             }
         }
     }
+
+    /// Check whenever `Codec::decode` method banned peer for message being too long.
+    /// If so bans peers
+    /// or returns the decoded message.
+    fn check_if_decoder_banned_peer(
```

Comment:
I'd rather not have this function extracted. There are three smells currently: 

* the function is named `check`, but it takes `&mut` and manipulates state. 
* it hides control-flow -- to understand the call-site, it's useful to see an `if` there
* it doesnt' actually hide contrlo flow -- we still have a match at the call-site, it's just that the logic is more obscured. 

The original version 

```rust
        let msg = match msg {
            Ok(msg) => msg,
            Err(ban_reason) => {
                self.ban_peer(ctx, ban_reason);
                return None;
            }
        };
```

looks perfectly readable to me: "the first thing we do is we ban the peer if we have to". 

In general, if you have the code shaped like

```rust
if complex_multiline_condition {
   complex_multiline_logic
}
```

the preferred simplification is 

```rust
let cond = self.compute_condition();
if cond {
   self.do_foo()
}
```

rather than

```rust
self.do_foo_optionally()
```

Keepin control flow at the call site makes for more readable code.

Reply:
@matklad 
This makes sense. However, we have methods like that, already:

```
    /// Hook called on every valid message received from this peer from the network.
    fn on_receive_message(&mut self) {
        if let Some(peer_id) = self.peer_id() {
            if self.last_time_received_message_update.elapsed()
                > UPDATE_INTERVAL_LAST_TIME_RECEIVED_MESSAGE
            {
                self.last_time_received_message_update = Clock::instant();
                self.peer_manager_addr.do_send(PeerManagerMessageRequest::PeerRequest(
                    PeerRequest::ReceivedMessage(peer_id, self.last_time_received_message_update),
                ));
            }
        }
    }```

We need to decide what to do with them.